test/functional/tests/fault_injection/test_fault_injection_opencas_load.py

```diff
@@ -0,0 +1,153 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+from time import sleep
+
+import pytest
+
+from api.cas import casadm, casadm_parser, cli, cli_messages
+from api.cas.cache_config import CacheMode, CleaningPolicy
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from core.test_run import TestRun
+from test_tools import fs_utils
+from test_tools.dd import Dd
+from test_tools.disk_utils import Filesystem
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+test_file_path = f"{mount_point}/test_file"
+
+
+@pytest.mark.parametrize("cache_mode", [CacheMode.WB, CacheMode.WO])
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_stop_no_flush_load_cache(cache_mode, filesystem):
+    """
+        title: Test to check that 'stop --no-data-flush' command works correctly.
+        description: |
+          Negative test of the ability of CAS to load unflushed cache on core device
+           with and without filesystem. Test uses write-back and write-only cache modes.
+        pass_criteria:
+          - No system crash while load cache.
+          - Starting cache without loading metadata fails.
+          - Starting cache with loading metadata finishes with success.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_part = core_dev.partitions[0]
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+
+    with TestRun.step("Change cleaning policy to NOP."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step(f"Add core with {filesystem.name} filesystem to cache and mount it."):
+        core_part.create_filesystem(filesystem)
+        core = cache.add_core(core_part)
+        core.mount(mount_point)
+
+    with TestRun.step(f"Create test file in mount point of exported object and check its md5 sum."):
+        test_file = fs_utils.create_random_test_file(f'{test_file_path}', Size(128, Unit.MebiByte))
```

Comment:
The purpose of this function in this place is to create some data on exported object. This would cause a dirty blocks in the cache. We would like to count this dirty blocks and check if its number is equal after stopping the cache without data flushing and load the cache again. If yes, than this option makes sense, if no, this test should fail.